scene/animation/animation_node_state_machine.cpp

```diff
@@ -1840,6 +1946,7 @@ void AnimationNodeStateMachine::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "state_machine_type", PROPERTY_HINT_ENUM, "Root,Nested,Grouped"), "set_state_machine_type", "get_state_machine_type");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "allow_transition_to_self"), "set_allow_transition_to_self", "is_allow_transition_to_self");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "reset_ends"), "set_reset_ends", "are_ends_reset");
+	ADD_PROPERTY(PropertyInfo(Variant::PACKED_STRING_ARRAY, "node_group_names", PROPERTY_HINT_ARRAY_TYPE, "NodeGroupNames", PROPERTY_USAGE_NO_EDITOR), "set_node_group_names", "get_node_group_names");
```

Comment:
Also, this line needs to be removed. Since serialization is supposed to be done by setter, so I think the registration here is for display in the inspector. Instead, an editor inspector plugin for StateMachine needs to be written. Please refer the `XXXInspectorPlugin` by searching `InspectorPlugin` in the core.

Reply:
I remember trying to remove this line before too. Which I did again, but doing so causes saving "node_group_names" fail silently, I guess this line is essential for creating a link to property or maybe just to Animation Node State Machine .tres file. Since other ones are part of node struct they are parsed in node_state_machine so they work a bit different. But I can also make this TOOLS_ENABLED, that works alright.

Reply:
It is not good to make this to TOOLS_ENABLED as it will appear in the class reference in spite of users can't use it in runtime. So `node_group_names` should be serialized in the `_set()` or in `_get_property_list()` like any other array property and serialized to tres.

Reply:
Does PropertyUsageFlag have an effect on this, for avoiding that, like PROPERTY_USAGE_INTERNAL or xx_SECRET or xx_NONE ? If now i guess it should be done in _get_property_list like the rest of those things..